From 87ba8a9ead4fd9e81ae3c5a894f698779de1f2c2
From: Vitali Lovich
Date: Wed, 16 Mar 2011 19:50:04 -0700
Subject: [PATCH] Linux: Detect and report that device has gone away while discarding URBs

Return LIBUSB_ERROR_NO_DEVICE if the device is no longer available
(ENODEV) to discard the urbs. References #82.
---
 libusb/os/linux_usbfs.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libusb/os/linux_usbfs.c b/libusb/os/linux_usbfs.c
--- a/libusb/os/linux_usbfs.c
+++ b/libusb/os/linux_usbfs.c
@@ -1401,6 +1401,9 @@ static int discard_urbs(struct usbi_transfer *itransfer, int first, int last_plu
 		if (EINVAL == errno) {
 			usbi_dbg("URB not found --> assuming ready to be reaped");
 			ret = LIBUSB_ERROR_NOT_FOUND;
+		} else if (ENODEV == errno) {
+			usbi_dbg("Device not found for URB --> assuming ready to be reaped");
+			ret = LIBUSB_ERROR_NO_DEVICE;
 		} else {
 			usbi_warn(TRANSFER_CTX(transfer),
 				"unrecognised discard errno %d", errno);
-- 
1.7.6.1
